zebra-chain/src/serialization/date_time.rs

```diff
@@ -34,6 +34,21 @@ impl DateTime32 {
             .try_into()
             .expect("unexpected out of range chrono::DateTime")
     }
+
+    /// If `earlier` is less than or equal to `self`, returns the number of
+    /// seconds elapsed between `earlier` and `self`.
+    ///
+    /// Otherwise, returns `None`.
+    pub fn checked_duration_since(&self, earlier: DateTime32) -> Option<u32> {
```

Comment:
`Duration`s have a nanosecond component. But this is a serialized type, so we want to store everything in terms of its serialized size (32 bit unsigned integer).

We could create a `Duration32` type. But `elapsed` is only used once, and `checked_duration_since` is only used in `elapsed`, so it's probably not worth it at this stage.

Reply:
Actually I think this is a good idea, because it helps simplify the code and calculations, and ensure they're correct using the type system.

See commit b49a734b. (There's a lot of boilerplate code, but the main change is removing a duration conversion to seconds.)

Reply:
(Also see commits 5a9ed6b4 and 69ce4129 in PR #2276, which show the value of this refactor for cleaning up code.)